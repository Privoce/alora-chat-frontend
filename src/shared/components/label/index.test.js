import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
	LabelComponent
} from 'shared/components';

describe('<LabelComponent />', () => {
	const modelI = {
		text: 'Label teste',
		danger: true,
		style: {
			fontSize: 10
		}
	};

	const getWrapper = (model) => {
		return shallow((
			<LabelComponent {...model} />
		));
	};

	it('renders correctly', () => {
		const wrapper = getWrapper(modelI);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('has a span component', () => {
		const wrapper = getWrapper(modelI);
		expect(wrapper.find('span').length).toBe(1);
	});

	it('span text content is equals modelI text', () => {
		const wrapper = getWrapper(modelI);
		expect(wrapper.find('span').text()).toBe(modelI.text);
	});
});
