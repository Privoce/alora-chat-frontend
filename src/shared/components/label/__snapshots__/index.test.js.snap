// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelComponent /> renders correctly 1`] = `
<span
  className="label fadeIn danger"
  style={
    Object {
      "fontSize": undefined,
      "margin": undefined,
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
  Label teste
</span>
`;
